### ISSUE TITLE
Missing MEDIA_LIBRARY permissions expo-media-library

### ISSUE THREAD

**Issue by Pietro-Putelli** [expo/expo#20625]
### Summary

I'm using `expo-media-library` to access user's media library and when ask permissions to user using 

`MediaLibrary.requestPermissionsAsync()` 

as response I got `status = "never_ask_again"` even if it never show up the popup for granting permissions.

So i tried to give permissions manually, through settings, but it occurs the same error:

`Missing MEDIA_LIBRARY permissions.`

Note that on the code down below I'm using react-native-permissions to ask for media library permissions, since the previous one does not show any popup for granting permissions.

### What platform(s) does this occur on?

Android

### Environment

  expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 12.6.1
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 16.14.2 - /usr/local/bin/node
      Yarn: 1.22.19 - ~/Desktop/Real/frontend/node_modules/.bin/yarn
      npm: 8.3.2 - /usr/local/bin/npm
    Managers:
      CocoaPods: 1.11.3 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
    IDEs:
      Android Studio: 2021.3 AI-213.7172.25.2113.9123335
      Xcode: 14.2/14C18 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^47.0.8 => 47.0.8
      react: ^18.2.0 => 18.2.0
      react-dom: ^18.2.0 => 18.2.0
      react-native: ^0.70.6 => 0.70.6
    npmGlobalPackages:
      eas-cli: 0.47.0
    Expo Workflow: bare

### Minimal reproducible example

```
import { PERMISSIONS, request } from "react-native-permissions";

export const requestMediaLibraryPermission = () => (dispatch) => {
  if (Platform.OS === "ios") {
    MediaLibrary.requestPermissionsAsync().then(({ status }) => {
      const isAuthorized = status == "granted";

      if (!isAuthorized) {
        Linking.openSettings();
      }
    });
  } else {
    request(PERMISSIONS.ANDROID.WRITE_EXTERNAL_STORAGE).then((status) => {
    });
  }
};
```

**Comment by brentvatne** [comment]
react-native-permissions may not play nicely with expo-media-library. if you can share a minimal reproducible example repository as described above, without including react-native-permissions, we can investigate!

**Comment by hamdij0maa** [comment]
Same issue, 
even after confirming the permissions from settings, 
the response of usePermissions in media library is always : granted = false;

**Comment by samkevin1** [comment]
Same issue here, the code below doesn't work:

```ts
import { printToFileAsync } from 'expo-print';
import { requestPermissionsAsync, createAssetAsync } from 'expo-media-library';

const print = async () => {
    try {
      const { uri } = await printToFileAsync({ html });
      const permission = await requestPermissionsAsync();

      if (permission.status === 'granted') {
        await createAssetAsync(uri);
      }
      
    } catch (error) {
      console.error(error);
    }
  };
```

When I log `permissions` this is what I get: `{"canAskAgain": false, "expires": "never", "granted": false, "status": "undetermined"}`. Granted is always false.

**Comment by VityaSchel** [comment]
No workaround/fix?

**Comment by FFsDeagle** [comment]
im getting this error too, its very frustrating. cant seem to fix it no matter what i do:

`    useEffect(() => {
        (async () => {
            const { status } = await MediaLibrary.requestPermissionsAsync()
            
            console.log(status)
            if (status === 'granted') {
                const media = await MediaLibrary.getAssetsAsync({
                    mediaType: 'photo',
                });
                console.log(media.assets);
            } else {
                handlePermissionDenied();
            }
        })();
      }, []);`

console.log(LOG  undetermined)

**Comment by Pietro-Putelli** [comment]
So any solution, because they close the errors and they don't even read that this is the biggest problem of the library.

**Comment by FFsDeagle** [comment]
I managed to fix the error, I updated expo and all my dependencies and it worked! 😁

**Comment by Pietro-Putelli** [comment]
export const requestMediaLibraryPermission = () => (dispatch) => {
  const errorCallback = (error) => {
    console.error(error);
  };

  const successCallback = (status) => {
    const isAuthorized = status == "granted";

    if (!isAuthorized) {
      Linking.openSettings();
    }

    dispatch(setPermissionsGranted({ media: isAuthorized }));
  };

  if (isAndroid) {
    PermissionsAndroid.request(PermissionsAndroid.PERMISSIONS.READ_MEDIA_IMAGES)
      .then(successCallback)
      .catch(errorCallback);
  } else {
    MediaLibrary.requestPermissionsAsync()
      .then(({ status }) => {
        successCallback(status);
      })
      .catch(errorCallback);
  }
};

**Comment by Pietro-Putelli** [comment]
> I managed to fix the error, I updated expo and all my dependencies and it worked! 😁

For me it doesn't work.

**Comment by FFsDeagle** [comment]
> > I managed to fix the error, I updated expo and all my dependencies and it worked! 😁
> 
> For me it doesn't work, can you be more specific on how you solved?
> Thank you very much

If i remember correctly i ran "expo update" in the terminal, which updated my expo sdk to the latest version. This process also updated all dependencies that i had in my project to suit the updated expo sdk.

It is reccomended that you create a backup of your project before running this, as it may cause errors.

What you also can do, before updating your projects sdk is start a new project and test it out there, as it will have the latest sdk. This is what i did, which is when i realized that i was using an old sdk.

**Comment by Pietro-Putelli** [comment]
Thank you, too much time spent on this issues I switched to react-native-camera-roll, it works as well.

**Comment by thebrrt** [comment]
Did you eject Expo? Encountering this same issue. Updated npx packages and refreshing Expo's cache didn't do anything for me. It's so weird. It works completely fine for iOS. 

I wonder if I'm supposed to be declaring permissions somewhere in app.json or something

**Comment by jaymes-bearden** [comment]
I have run in to this issue for Android and got a working solution for my circumstances:
- Expo SDK47 (ejected)
- expo-media-library ~15.3.0

For Android < 13 this issue does not happen -- only on Android >= 13. This issue occurred when attempting a `MediaLibrary.saveToLibraryAsync(..)` despite `MediaLibrary.requestPermissionsAsync` reporting status as `granted` and boolean granted=true.

For my situation, I only needed write access, so I made the request as:
`const permission = await MediaLibrary.requestPermissionsAsync(true);`

With this, all versions of Android would report the access as granted BUT, attempts to actually use a permission gated function would fail on Android 13.

With this changed to:
`const permission = await MediaLibrary.requestPermissionsAsync();`

I'm able to used permission gated functions correctly.

NOTE: This appears to be fixed for Android in expo-media-library >= 15.3.0 but the IOS builds will fail due to an implicit dependency on SDK48.

**Comment by clevadani** [comment]
If you're still experiencing this on android, please run `npx expo-cli upgrade`
